icu4c/source/i18n/collationtailoring.h

```diff
@@ -105,6 +107,10 @@ struct U_I18N_API CollationCacheEntry : public SharedObject {
     const CollationTailoring *tailoring;
 };
 
+template<> U_I18N_API
+const CollationCacheEntry *
+LocaleCacheKey<CollationCacheEntry>::createObject(const void *creationContext,
+                                                  UErrorCode &errorCode) const;
```

Comment:
Pardon my ignorance, but where is this function actually implemented?  What's going on here?

I see this happens in several other classes below, so the same question applies to all of them.  Is the idea that this doesn't actually _need_ to be implemented, but it _does_ need to be declared?  Or is the implementation getting inherited from somewhere?

Reply:
Template specializations essentially tell the compiler that the code defines its own implementation for a specific type parameter (unlike template instantiations). This function itself is defined and exposed in https://github.com/unicode-org/icu/blob/main/icu4c/source/i18n/ucol_res.cpp#L139 

For a tool like TAPI, that attempts to discover all linkable symbols from headers, this function looks hidden. Declaring this function template specialization now resolves this. 

I thought it was reasonable to put these specializations next to the class declaration for the type parameter, in this case `CollationCacheEntry`.

Reply:
Thanks for the pointers here; I don't know how I missed that function.  Obviously I was searching for the wrong thing.  Thanks for the explanation; your actual changes look fine.